agent/src/softwarecontaineragent.cpp

```diff
@@ -91,6 +94,47 @@ SoftwareContainerAgent::~SoftwareContainerAgent()
 {
 }
 
+/* we are fetching lxcpath from config now. But in near future we should use lxcConfigPath instead
+ * */
+void SoftwareContainerAgent::removeOldContainers()
+{
+    char **containerNames = nullptr;
+    struct lxc_container **containerList = nullptr;
+
+    const char *basePath = lxc_get_global_config_item("lxc.lxcpath");
+    auto num = list_all_containers(basePath, &containerNames, &containerList);
+    if (num > 0 && num < std::numeric_limits<int>::max()) {
+        log_warning() << num << " unused deprecated containers found";
+        for (; num; num--) {
```

Comment:
This type of truth checking is not nice. I would prefer a regular for loop with a new variable which you can compare to num. That way, you also can get rid of the `if (num > 0)` check above I guess.

TL;DR: Numbers are not bools :)